src/lib/auth/authconf.rb

```diff
@@ -475,12 +475,18 @@ def ldap_export
 
         # Set configuration for LDAP from exported objects.
         def ldap_import(exported_conf)
-            @ldap_conf = exported_conf['conf']
-            @ldap_conf = {} if @ldap_conf.nil?
-            @ldap_pam = exported_conf['pam']
-            @ldap_pam = false if @ldap_pam.nil?
-            @ldap_nss = exported_conf['nss']
-            @ldap_nss = [] if @ldap_nss.nil?
+            if exported_conf.nil?
+                @ldap_conf = {}
+                @ldap_pam = false
+                @ldap_nss = []
+            else
+                @ldap_conf = exported_conf['conf']
+                @ldap_conf = {} if @ldap_conf.nil?
+                @ldap_pam = exported_conf['pam']
+                @ldap_pam = false if @ldap_pam.nil?
+                @ldap_nss = exported_conf['nss']
+                @ldap_nss = [] if @ldap_nss.nil?
+            end
```

Comment:
NP / Suggestion: pieces of code like this could be _simplified_ by having a `default_conf` and using the [`#merge` Hash method](https://ruby-doc.org/core-2.7.0/Hash.html#method-i-merge).

For example, 

```ruby

...

def ldap_import(exported_conf)
  default_ldap_conf.merge(Hash(exported_conf))
end

...

def default_ldap_conf
  @default_ldap_conf ||= { 'conf' => {}, 'pam' => false, 'nss' => [] }
end
```

Testing it in `irb` 

```irb
irb(main):007:1* def default_ldap_conf
irb(main):008:1*   @default_ldap_conf ||= { 'conf' => {}, 'pam' => false, 'nss' => [] }
irb(main):009:0> end
=> :default_ldap_conf

irb(main):010:1* def ldap_import(exported_conf)
irb(main):011:1*   default_ldap_conf.merge(Hash(exported_conf))
irb(main):012:0> end
=> :ldap_import

irb(main):013:0> exported_conf = nil
=> nil
irb(main):014:0> ldap_import(exported_conf)
=> {"conf"=>{}, "pam"=>false, "nss"=>[]}

irb(main):015:0> exported_conf = { 'conf' => { something: :here }, 'pam' => true }
=> {"conf"=>{:something=>:here}, "pam"=>true}
irb(main):016:0> ldap_import(exported_conf)
=> {"conf"=>{:something=>:here}, "pam"=>true, "nss"=>[]}

```

Reply:
Yes, changes were manually tested.

As I said in #100, thank you for the suggestions, I will put them into practice in the future.